libcontainer/seccomp/seccomp_linux.go

```diff
@@ -110,6 +110,21 @@ func InitSeccomp(config *configs.Seccomp) (int, error) {
 		}
 	}
 
+	// Enable libseccomp binary tree optimization for longer rulesets.
+	//
+	// The number below chosen semi-arbitrarily, considering the following:
+	// 1. libseccomp <= 2.5.4 misbehaves when binary tree optimization
+	// is enabled and there are 0 rules.
+	// 2. All known libseccomp versions (2.5.0 to 2.5.4) generate a binary
+	// tree with 4 syscalls per node.
+	if len(config.Syscalls) > 32 {
+		err = filter.SetOptimize(2)
+		if err != nil {
```

Comment:
minor nit; wondering why you didn't make this a local variable (it made me look at code further down and up to check if this was used in a `defer` or if there was a special reason), i.e., this could be;

```go
if err := filter.SetOptimize(2); err != nil {
```

Reply:
Fixed; PTAL @thaJeztah